backend-rust/src/graphql_api/baker.rs

```diff
@@ -137,11 +147,118 @@ impl Baker {
         Account::query_by_index(get_pool(ctx)?, i64::from(self.id)).await?.ok_or(ApiError::NotFound)
     }
 
-    // transactions("Returns the first _n_ elements from the list." first: Int
-    // "Returns the elements in the list that come after the specified cursor."
-    // after: String "Returns the last _n_ elements from the list." last: Int
-    // "Returns the elements in the list that come before the specified cursor."
-    // before: String): BakerTransactionRelationConnection
+    async fn transactions(
+        &self,
+        ctx: &Context<'_>,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
+        before: Option<String>,
+    ) -> ApiResult<connection::Connection<String, InterimTransaction>> {
+        let config = get_config(ctx)?;
+        let pool = get_pool(ctx)?;
+        let query = ConnectionQuery::<i64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.transactions_per_block_connection_limit,
+        )?;
+
+        // Retrieves the transactions related to a baker account ('AddBaker',
+        // 'RemoveBaker', 'UpdateBakerStake', 'UpdateBakerRestakeEarnings',
+        // 'UpdateBakerKeys', 'ConfigureBaker'). The transactions are ordered in
+        // descending order (outer `ORDER BY`). If the `last` input parameter is
+        // set, the inner `ORDER BY` reverses the transaction order to allow the
+        // range be applied starting from the last element.
+        let mut row_stream = sqlx::query_as!(
+            Transaction,
+            r#"
+            SELECT * FROM (
+                SELECT 
+                    index,
+                    block_height,
+                    hash,
+                    ccd_cost,
+                    energy_cost,
+                    sender_index,
+                    type as "tx_type: DbTransactionType",
+                    type_account as "type_account: AccountTransactionType",
+                    type_credential_deployment as "type_credential_deployment: CredentialDeploymentTransactionType",
+                    type_update as "type_update: UpdateTransactionType",
+                    success,
+                    events as "events: sqlx::types::Json<Vec<Event>>",
+                    reject as "reject: sqlx::types::Json<TransactionRejectReason>"
+                FROM transactions
+                WHERE transactions.sender_index = $5
+                AND index > $1 AND index < $2
+                AND type_account IN ('AddBaker', 'RemoveBaker', 'UpdateBakerStake', 
```

Comment:
Where is this list obtained from?

Can we somehow map enum types into this list?
And if not could we reuse the names provided as part of TransactionType? I assume it would be those which are used to generate the type.

Reply:
The advantage of doing it this way is that if someone renames the enums, then it will still work.

Reply:
The enum is is a subset of the all possible `account_transaction_type`:

https://github.com/Concordium/concordium-scan/blob/main/backend-rust/src/migrations/m0001-initialize.sql#L3

Reply:
compile-time checked queries would catch this